### PR TITLE
Refactor filter drivers

### PR DESCRIPTION
**Pull request by waltermwaniki**
This PR updates the filter/filterwheel device type.  
- Individual filters are exposed as separate object instances rather than exposing a filterwheel with separate filter settings. Filter instances require a filterwheel instance when they are initialized.

### CODE REVIEW

**Comment by adamkglaser** [comment]
This looks good to me thanks. Micah is this API behavior similar to how you currently implemented things?

File: voxel/devices/filter/asi.py [hunk 1]
```diff
@@ -1,18 +1,89 @@
...
+    def disable(self) -> None:
+        """Disable this filter if it's active."""
+        if self.is_active:
+            self.wheel._current_filter = None
 
-    def enable(self):
-        """Set parent filter wheel to filter"""
-        self.wheel.filter = self.id
+    @property
+    def is_active(self) -> bool:
```
Comment (micahwoodard) [inline voxel/devices/filter/asi.py:84]: maybe change name to enabled?

Comment (WalterMwaniki) [reply voxel/devices/filter/asi.py]: I updated the method name to use `def enabled()`

Comment (WalterMwaniki) [reply voxel/devices/filter/asi.py]: @micahwoodard, @adamkglaser what do you think about requiring that the current filter is disabled before activating a new one. This is opposed to automatically disabling a filter if a different one on the same wheel is enabled.

Comment (micahwoodard) [reply voxel/devices/filter/asi.py]: To me it seems like an unnecessary step.  Is there a scenario where this is advantageous?

**Pull request closed without merging**